internal/ir/infra.go

```diff
@@ -38,6 +37,14 @@ type ProxyInfra struct {
 	Listeners []ProxyListener
 }
 
+// InfraMetadata defines metadata for the managed proxy infrastructure.
+// +k8s:deepcopy-gen=true
+type InfraMetadata struct {
+	// Labels define a map of string keys and values that can be used to organize
+	// and categorize proxy infrastructure objects.
+	Labels map[string]string
```

Comment:
thoughts on `type InfraMetadata map[string]string` ?

Reply:
IMHO the current approach makes it easier to introduce additional metadata types in the future, i.e. `ResourceVersion`, `CreationTimestamp`, etc.